UIKitTests/UIScrollViewTests.swift

```diff
@@ -47,6 +47,65 @@ class UIScrollViewTests: XCTestCase {
         XCTAssertEqual(scrollView.contentInset, arbitraryContentInset)
     }
 
+    func testScrollIndicatorsViewHierarchyPositions() {
+        // subviews[0] is the back-most view, higher indexes are higher in layer hierarchy
+        // we want scroll indicators to always remain on top of 'normal' subviews
+        // and we want newer 'normal' subviews to be above older ones
+
+        // TODO: iOS does not allow explicit access to scroll indicators via subviews
+        // we should remove that, too - but for now it's the easy way to go and helpful in testing
+
+        let view1 = UIView()
+        let view2 = UIView()
+        let view3 = UIView()
+
+        scrollView.addSubview(view1)
+        scrollView.addSubview(view2)
+        scrollView.addSubview(view3)
+
+        let horizontalIndicatorIndex = scrollView.subviews.index(of: scrollView.horizontalScrollIndicator)!
+        let verticalIndicatorIndex = scrollView.subviews.index(of: scrollView.verticalScrollIndicator)!
+        let label1Index = scrollView.subviews.index(of: view1)!
+        let label2Index = scrollView.subviews.index(of: view2)!
+        let label3Index = scrollView.subviews.index(of: view3)!
+
+        XCTAssert(horizontalIndicatorIndex > label1Index)
+        XCTAssert(verticalIndicatorIndex > label1Index)
+
+        XCTAssert(label2Index > label1Index)
+
+        XCTAssertEqual(label1Index, 0)
```

Comment:
what's the point in adding three subviews? we already have other tests that test the behaviour of `addSubview`. we're just writing more test code that we have to maintain.

how is this different to the case at the bottom?